gbn/queue.go

```diff
@@ -112,36 +127,48 @@ func (q *queue) resend(cb func(packet *PacketData) error) error {
 	q.topMtx.RUnlock()
 
 	if base == top {
+		q.syncer.reset()
```

Comment:
why do we need a `reset` here? 

Initially in my diff I had it here cause there was an `init` state above this. but if that is removed then this can be removed right?

Reply:
btw - the reason i had the other "init" state here was to remove the need for locking the awaitMu mutex before doing the seqBase and seqTop locks. 

if we still want there to be no other "resend" call during the time this is going, then we should add that back. 

but i think this is synchronous for now right?

Reply:
> why do we need a reset here

Ah my bad, missed removing that. Thanks!

> btw - the reason i had the other "init" state here was to remove the need for locking the awaitMu mutex before doing the seqBase and seqTop locks.

Yeah, it's synchronous currently, so we will never call the `queue.resend` function twice at the same time while it's currently resending the function. 
Since we now process the awaited ACK+NACK while we are also resending packets, we now need to ensure that we set the `syncStateResending` while under the same lock that we also set the `awaitedACK` + `awaitedNACK` which is why I changed this. But like you say, if we add an extra init state to the syncer we could achieve that we wouldn't allow two concurrent calls of the `queue.resend` function. Opted to not add it now though to not cause extra confusion with unused functionality, butcan change it if you'd prefer :).